p2p/protocol/identify/id.go

```diff
@@ -432,11 +443,34 @@ func (ids *IDService) getSnapshot() *identifySnapshot {
 	return snapshot
 }
 
-func (ids *IDService) populateMessage(
-	mes *pb.Identify,
+func (ids *IDService) writeChunkedIdentifyMsg(c network.Conn, snapshot *identifySnapshot, s network.Stream) error {
+	mes := ids.getIdentifyMsgUnsigned(c, snapshot)
+	sr := ids.getSignedRecord(snapshot)
+	mes.SignedPeerRecord = sr
+	writer := ggio.NewDelimitedWriter(s)
+
+	if sr == nil || proto.Size(mes) <= legacyIDSize {
+		return writer.WriteMsg(mes)
+	} else {
+		mes.SignedPeerRecord = nil
+		mes.SignedRecordInNextMessage = true
+		if err := writer.WriteMsg(mes); err != nil {
+			return err
+		}
+
+		// then write just the signed record
+		m := &pb.Identify{SignedPeerRecord: sr}
```

Comment:
Can't the signed peer record be `nil`?

Reply:
@raulk We already check for that at https://github.com/libp2p/go-libp2p/blob/3e1b364fefea98ba241e8774529bf5bb49a3cea7/p2p/protocol/identify/id.go#L452.